tensorflow_io/core/kernels/audio_kernels.cc

```diff
@@ -19,8 +19,86 @@ limitations under the License.
 
 namespace tensorflow {
 namespace data {
+
+// DecodedAudio
+size_t DecodedAudio::data_size() {
+  return channels * samples_perchannel * sizeof(int16);
+}
+
+DecodedAudio::DecodedAudio(bool success, size_t channels,
+                           size_t samples_perchannel, size_t sampling_rate,
+                           int16 *data)
+    : success(success), channels(channels),
+      samples_perchannel(samples_perchannel), sampling_rate(sampling_rate),
+      data(data) {}
+
+DecodedAudio::~DecodedAudio() {
+  if (data) {
+    std::free((void *)data);
+  }
+}
+
+// DecodeAudioBaseOp
+DecodeAudioBaseOp::DecodeAudioBaseOp(OpKernelConstruction *context) : OpKernel(context) {}
+
+void DecodeAudioBaseOp::Compute(OpKernelContext *context) {
```

Comment:
I'm a bit unhappy with the fact that these methods are in `tensorflow::data` while everything else is in this nested nameless namespace. I'm not experienced enough with C++ to know what this is about, so happy about feedback/ideas.